Checkout: Adjust spacing between override list items and area behind flag

sirbrillig opened the pull request:
## Proposed Changes

This is a follow-up to https://github.com/Automattic/wp-calypso/pull/83890. This PR adjusts the styles of the list of cost overrides so that they match the design.

The annotations in the first screenshot below show the requested design changes that this PR implements (this is a screenshot _without_ the changes; screenshots with the changes are below):

![283992714-4b38c3a7-1935-4b13-a361-c0be2cf26e40](https://github.com/Automattic/wp-calypso/assets/2036909/b0dfc0a0-f616-4b6d-932e-0c6923ec2019)

> [!NOTE] 
> The redesigned checkout remains hidden behind both a feature flag and a query string parameter. It is only available currently in development (`calypso.localhost`) and on `wpcalypso` which is used by the `calypso.live` branches below.

Before             |  After
:-------------------------:|:-------------------------:
<img width="262" alt="Screenshot 2023-11-19 at 1 42 21 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/c810a8d1-a7cf-49ca-b3bf-1e4cd7a9f099"> | <img width="256" alt="Screenshot 2023-11-20 at 11 03 34 AM" src="https://github.com/Automattic/wp-calypso/assets/2036909/d678f2c7-7141-49af-aa40-c78bd1dd6802">

This is part of the project to implement a redesign of checkout: https://github.com/Automattic/payments-shilling/issues/1969

## Testing Instructions

- Use a site with an existing plan.
- Add a plan upgrade (a higher level plan) to your cart and visit checkout.
- Add `?checkoutVersion=2` to the end of the checkout URL and reload checkout.
- Verify that the sidebar spacing matches the design instruction shown above.

sirbrillig committed:
Adjust spacing between override list items and area

sirbrillig committed:
Remove top margin of first override

sirbrillig commented:
@JanaMW27 I implemented the spacing from your comment here: https://github.com/Automattic/wp-calypso/pull/83890#issuecomment-1817466855. Does this look right?

JanaMW27 commented:
Hello, the 16 px spacing is only for the COUPON to the rest of the discounts (the spacing between the non-coupon discounts is good in the Before one)

Also, could we do the chevron 16px x 16 px on the billing cycle dropdown?

Sorry for all the little changes, and super thank you!

sirbrillig committed:
Only apply margin to coupon override

sirbrillig commented:
Thank you, @JanaMW27! That looks much better. I've updated the code and the screenshot.

> Also, could we do the chevron 16px x 16 px on the billing cycle dropdown?

I've added that task to our tracking issue (https://github.com/Automattic/payments-shilling/issues/1969#issuecomment-1817529355). I have a bunch of things to edit on those dropdowns.

sirbrillig committed:
Restore original 4px margin for normal items

jjchrisdiehl commented:
I think you might have pointed this out in an earlier iteration of our redesign, and I'm not sure if we're accounting for it elsewhere, but the label and price don't appear to line up here:

![image](https://github.com/Automattic/wp-calypso/assets/16580129/30302c2b-a0a0-4254-b401-a93f924dfeb1)

It looks like the line-height for both of these items is being calculated slightly differently. The label is 21px while the price is 26.4. Both derive from the `body` tag which is set to "1.5" but no unit is applied. Not sure why they diverge but that seems to create the misalignment.

Otherwise, everything else looks good except for the coupon code line item:

![image](https://github.com/Automattic/wp-calypso/assets/16580129/3db97c1a-fdef-4291-852b-5bea8dc5313f)

While this matches the Figma, it doesn't seem to match what you have in the PR description:

| Figma | PR |
| ------ | -------|
| ![image](https://github.com/Automattic/wp-calypso/assets/16580129/4f59f7a8-95e5-470d-a93f-5219688df8c0) | ![image](https://github.com/Automattic/wp-calypso/assets/16580129/a3c98e79-d1fb-4469-93c9-493cbec217c6) |

Unless the drawn in values are what we are looking for rather than what the image depicts?

JanaMW27 commented:
> I think you might have pointed this out in an earlier iteration of our redesign, and I'm not sure if we're accounting for it elsewhere, but the label and price don't appear to line up here:

The product name and the price should have the same typography (size, color, alignment)
<img width="520" alt="Screenshot 2023-11-21 at 10 34 40" src="https://github.com/Automattic/wp-calypso/assets/71137829/8fbb6210-14f5-4979-9789-557ff6af8475">

> Unless the drawn in values are what we are looking for rather than what the image depicts?

Yes, the annotated values are what they should be (following what I have on the design) not what we currently have.

I hope this makes things more clear, we can connect in slack to chat more about it if you need more info.

jjchrisdiehl commented:
> As Jana said, the numbers she drew on top of the old screenshot are what we are looking for, and that's what this PR provides. I updated the PR description to clarify that the first screenshot is of the UI _without_ the changes.

Ah good to know, then in that case it looks good!
